bridged-nft-token/src/nft_core.rs

```diff
@@ -0,0 +1,289 @@
+    ) -> bool;
+}
+
+trait NonFungibleTokenResolver {
+    fn nft_resolve_transfer(
+        &mut self,
+        owner_id: AccountId,
+        receiver_id: AccountId,
+        approved_account_ids: HashMap<AccountId, U64>,
+        token_id: TokenId,
+    ) -> bool;
+}
+
+#[near_bindgen]
+impl NonFungibleTokenCore for Contract {
+    #[payable]
+    fn nft_transfer(
+        &mut self,
+        receiver_id: ValidAccountId,
+        token_id: TokenId,
+        approval_id: Option<U64>,
+        memo: Option<String>,
+    ) {
+        assert_one_yocto();
+
+        let sender_id = env::predecessor_account_id();
+        let previous_token = self.internal_transfer(
+            &sender_id,
+            receiver_id.as_ref(),
+            &token_id,
+            approval_id,
+            memo,
+        );
+        refund_approved_account_ids(
+            previous_token.owner_id,
+            &previous_token.approved_account_ids,
+        );
+    }
+
+    #[payable]
+    fn nft_transfer_call(
+        &mut self,
+        receiver_id: ValidAccountId,
+        token_id: TokenId,
+        approval_id: Option<U64>,
+        memo: Option<String>,
+        msg: String,
+    ) -> Promise {
+        assert_one_yocto();
+        let sender_id = env::predecessor_account_id();
+        let previous_token = self.internal_transfer(
+            &sender_id,
+            receiver_id.as_ref(),
+            &token_id,
+            approval_id,
+            memo,
+        );
+        // Initiating receiver's call and the callback
+        ext_non_fungible_token_receiver::nft_on_transfer(
+            sender_id,
+            previous_token.owner_id.clone(),
+            token_id.clone(),
+            msg,
+            receiver_id.as_ref(),
+            NO_DEPOSIT,
+            env::prepaid_gas() - GAS_FOR_NFT_TRANSFER_CALL,
+        )
+            .then(ext_self::nft_resolve_transfer(
+                previous_token.owner_id,
+                receiver_id.into(),
+                previous_token.approved_account_ids,
+                token_id,
+                &env::current_account_id(),
+                NO_DEPOSIT,
+                GAS_FOR_RESOLVE_TRANSFER,
+            ))
+    }
+
+    #[payable]
+    fn nft_approve(&mut self, token_id: TokenId, account_id: ValidAccountId, msg: Option<String>) {
+        assert_at_least_one_yocto();
+        let account_id: AccountId = account_id.into();
+
+        let mut token = self.tokens_by_id.get(&token_id).expect("Token not found");
+
+        assert_eq!(
+            &env::predecessor_account_id(),
+            &token.owner_id,
+            "Predecessor must be the token owner."
+        );
+
+        let approval_id: U64 = token.next_approval_id.into();
+        let is_new_approval = token
+            .approved_account_ids
+            .insert(account_id.clone(), approval_id)
+            .is_none();
+
+        let storage_used = if is_new_approval {
+            bytes_for_approved_account_id(&account_id)
```

Comment:
If I recall correctly, there are some bytes missing here, that come from some overhead in the trie.

Why not estimating the storage used by checking storage before and after?

Reply:
it was removed and replaced by the one contained inside the standard